### ISSUE TITLE
[FEA] SubwordTokenizer to support QA type inputs

### ISSUE THREAD

**Issue by chloeh13q** [rapidsai/cudf#9520]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I wish I could use cuDF to do [...]

I wish to use cuDF to preprocess QA type inputs, specifically fine-tuning BERT on SQuAD for question answering. The tokenization in the SQuAD dataset is different from how BERT tokenizes the input.
The `SubwordTokenizer` does not currently support sentence pairs as input or return an offset mapping. (I can potentially get around the first issue by stitching together sentence pairs post-tokenization.)

**Describe the solution you'd like**

Huggingface does this by offering the option of `return_offsets_mapping`. It would be helpful to have the same option in `SubwordTokenizer`.

**Describe alternatives you've considered**

I've considered adding additional steps to complement the `SubwordTokenizer` (e.g. after tokenization, loop over the characters again and generate an offset mapping separately, on CPU). However, by doing so I may lose much of the GPU acceleration. I can also give up on using a GPU-accelerated tokenizer completely.

**Additional context**
Add any other context, code examples, or references to existing implementations about the feature request here.
- [Huggingface's `PreTrainedTokenizerBase.__call__`](https://huggingface.co/transformers/internal/tokenization_utils.html#transformers.tokenization_utils_base.PreTrainedTokenizerBase.__call__)
- [Example notebook for fine-tuning BERT for QA using Huggingface](https://github.com/huggingface/notebooks/blob/master/examples/question_answering.ipynb)

**Comment by beckernick** [comment]
@VibhuJawa would something like this potentially require work in cuDF Python or libcudf?

**Comment by VibhuJawa** [comment]
> @VibhuJawa would something like this potentially require work in cuDF Python or libcudf?

This will require `libcudf` work for us to get a clean implementation across parameters. 

We can get a  python implementation working but stuff like  `max-length`, `padding` and `stride`  will not work cleanly there without `libcudf` changes.

**Comment by davidwendt** [comment]
Is this still needed?
@VibhuJawa Would you be able to specify what changes are needed in libcudf for this?

**Comment by VibhuJawa** [comment]
> Is this still needed? @VibhuJawa Would you be able to specify what changes are needed in libcudf for this?
I dont know of any customer use cases right now for it. @beckernick, do you know about any

**Comment by vyasr** [comment]
Based on the above threads I'm going to close this as a wontfix. Happy to reopen if we see future demand for it.
